solr/core/src/java/org/apache/solr/handler/IndexFetcher.java

```diff
@@ -321,9 +369,16 @@ boolean fetchLatestIndex(boolean forceReplication, boolean forceCoreReload) thro
       try {
         response = getLatestVersion();
       } catch (Exception e) {
-        LOG.error("Master at: " + masterUrl + " is not available. Index fetch failed. Exception: " + e.getMessage());
-        return false;
-      }
+        final String errorMsg = e.getMessage();
```

Comment:
Granted e.getMessage() was what this code was doing before you made changes but can you please use e.toString() instead?  IMO, e.getMessage() should rarely if ever be called instead of e.toString() because e.toString() critically contains the name of the exception itself.  The "message" alone can be unclear without the exception class.  Consider `FileNotFoundException` and there are others.